br/pkg/task/backup_ebs.go

```diff
@@ -253,6 +273,124 @@ func RunBackupEBS(c context.Context, g glue.Glue, cmdName string, cfg *BackupEBS
 	return nil
 }
 
+func waitAllScheduleStoppedAndNoRegionHole(ctx context.Context, cfg Config, mgr *conn.Mgr) error {
+	allStores, err := conn.GetAllTiKVStoresWithRetry(ctx, mgr.GetPDClient(), util.SkipTiFlash)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	// we wait for 15*40 = 600s = 10m at least
+	for retryIdx := 0; retryIdx < 40; retryIdx++ {
+		if ctx.Err() != nil {
+			return ctx.Err()
+		}
+		if retryIdx > 0 {
+			time.Sleep(waitAllScheduleStoppedInterval)
+		}
+		allRegions, err2 := waitUntilAllScheduleStopped(ctx, cfg, allStores, mgr)
+		if err2 != nil {
+			if causeErr := errors.Cause(err2); causeErr == hasPendingAdminErr {
+				log.Info("schedule ongoing on tikv, will retry later", zap.Error(err2))
+			} else {
+				log.Warn("failed to wait schedule, will retry later", zap.Error(err2))
+			}
+			continue
+		}
+
+		log.Info("all leader regions got, start checking hole", zap.Int("len", len(allRegions)))
+		// sort by start key
+		sort.Slice(allRegions, func(i, j int) bool {
+			left, right := allRegions[i], allRegions[j]
+			return bytes.Compare(left.StartKey, right.StartKey) < 0
+		})
+
+		var hasHole bool
+		for j := 0; j < len(allRegions)-1; j++ {
+			left, right := allRegions[j], allRegions[j+1]
+			// todo: do we need to prefix every key with 'z'?, seems doesn't affect sorting and checking
+			if bytes.Compare(left.EndKey, right.StartKey) != 0 {
```

Comment:
can we handle special end key like `""` here?

Reply:
we sort by start key and we don't check the end key of the region sorted last.

Reply:
add comments here https://github.com/pingcap/tidb/pull/37739/commits/0c02f826ef0d4f97c383f5e77a6bb2c32351d68c